dipy/core/sphere.py

```diff
@@ -498,6 +500,155 @@ def disperse_charges(hemi, iters, const=.2):
+    vects = (vects.T / np.sqrt((vects ** 2).sum(1))).T
+    grad = np.zeros((N, N * 3))
+    for i in range(3):
+        grad[:, i * N:(i + 1) * N] = np.diag(vects[:, i])
+    return grad
+
+
+def _get_forces_alt(vects, alpha=2.0, **kwargs):
+    """Electrostatic-repulsion objective function. The alpha parameter
+    controls the power repulsion (energy varies as $1 / r^\alpha$).
+
+    Parameters
+    ----------
+    vects : array-like (N * 3,)
+        Points on the sphere.
+    alpha : float
+        Controls the power of the repulsion. Default is 1.0.
+    weights : array-like (N, N)
+        Weight values to the electrostatic energy.
+
+    Returns
+    -------
+    energy : float
+        Sum of all interactions between any two vectors.
+
+    """
+
+    nb_points = vects.shape[0] / 3
+    # weights = kwargs.get('weights', np.ones((nb_points, nb_points)) / nb_points**2)
+    weights = kwargs.get('weights', np.ones((nb_points, nb_points)))
+    charges = vects.reshape((nb_points, 3))
+    all_charges = np.concatenate((charges, -charges))
+    all_charges = all_charges[:, None]
+    r = charges - all_charges
+    r_mag = np.sqrt((r * r).sum(-1))[:, :, None]
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        potential = 1 / r_mag ** alpha
+
+    d = np.arange(len(charges))
+    potential[d, d] = 0
+    potential = potential[:nb_points] + potential[nb_points:]
+    potential = weights * potential.sum(-1)
+    potential = potential.sum()
+    return potential
+
+
+def _get_grad_forces_alt(vects, alpha=2.0, **kwargs):
+    """1st-order derivative of electrostatic-like repulsion energy.
+
+    Parameters
+    ----------
+    vects : array-like (N * 3,)
+        Points on the sphere.
+    alpha : float
+        Controls the power of the repulsion. Default is 1.0.
+
+    Returns
+    -------
+    grad : array-like (N * 3,)
+        Gradient of the objective function.
+
+    """
+
+    nb_points = vects.shape[0] / 3
+    # weights = kwargs.get('weights', np.ones((nb_points, nb_points)) / nb_points**2)
+    weights = kwargs.get('weights', np.ones((nb_points, nb_points)))
+    charges = vects.reshape((nb_points, 3))
+    all_charges = np.concatenate((charges, -charges))
+    all_charges = all_charges[:, None]
+    r = charges - all_charges
+    r_mag = np.sqrt((r * r).sum(-1))[:, :, None]
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        forces = -2 * alpha * r / r_mag ** (alpha + 2.)
+
+    d = np.arange(len(charges))
+    forces[d, d] = 0
+    forces = forces[:nb_points] + forces[nb_points:]
+    forces = forces * weights.reshape((nb_points, nb_points, 1))
+    forces = forces.sum(0)
+    return forces.reshape((nb_points * 3))
+
+
+def disperse_charges_alt(init_pointset, iters, tol=1.0e-3):
+    """Reimplementation of disperse_charges making use of
+    scipy.optimize.fmin_slsqp.
+
+    Parameters
+    ----------
+    init_pointset : HemiSphere
```

Comment:
Technically here I think we are expecting a numpy array, right? Can you update the docstring?

Reply:
True. Thanks. Fixed in b693c60.